pyccel/stdlib/ndarrays/ndarrays.h

```diff
@@ -0,0 +1,66 @@
+#ifndef NDARRAYS_H
+# define NDARRAYS_H
+
+# include <stdlib.h>
+# include <complex.h>
+# include <string.h>
+# include <stdio.h>
+# include <stdarg.h>
+
+typedef union	u_ndarr_type
+{
+	/*
+	*** data --- change it later to shorter names or use anonymous array
+	*/
+	char	*raw_data;
+	int		*int_nd;
+	float	*float_nd;
+	double	*double_nd;
+    double complex *complex_double;
+}				t_ndarray_type;
+
+typedef struct s_slice_data{
+	int start;
+	int end;
+	int step;
+}	t_slice;
+
+typedef struct s_ndarray
+{
+	/* raw data buffer*/
+	t_ndarray_type *data;
```

Comment:
Why does this need to be a pointer? Can we have a `t_ndarray` with `data==NULL` or where `data` contains multiple objects of type `t_ndarray_type`?

Reply:
will resolve when fix the pointers for everything

Reply:
i did change it to an anonymous union